creator-node/src/diskManager.test.js

```diff
@@ -0,0 +1,93 @@
+    const validPath = path.join(DiskManager.getConfigStoragePath(), 'files', 'muU', 'QmYfSQCgCwhxwYcdEwCkFJHicDe6rzCAb7AtLz3GrHmuU6')
+    assert.deepStrictEqual(fullPath, validPath)
+  })
+
+  it('Should fail if fileName is not passed into computeBasePath', function () {
+    try {
+      DiskManager.computeBasePath()
+    } catch (e) {
+      assert.ok(e.message.includes('Please pass in a valid fsDest to computeBasePath'))
+    }
+  })
+
+  it(`Should fail if fileName doesn't contain the appropriate amount of characters`, function () {
+    try {
+      DiskManager.computeBasePath('asd')
+    } catch (e) {
+      assert.ok(e.message.includes('Please pass in a valid fsDest to computeBasePath'))
+    }
+  })
+
+  it(`Should fail if fileName contains a slash`, function () {
+    try {
+      DiskManager.computeBasePath('/file_storage/asdf')
+    } catch (e) {
+      assert.ok(e.message.includes('Cannot pass in a directory path into this function, please pass in the leaf dir or file name'))
+    }
+  })
+
+  /**
+   * computeFilePathInDir
+   */
+  it('Should pass if computeBasePath returns the correct path', function () {
+    const fullPath = DiskManager.computeFilePathInDir('QmdirectoryName', 'QmYfSQCgCwhxwYcdEwCkFJHicDe6rzCAb7AtLz3GrHmuU6')
+    const validPath = path.join(DiskManager.getConfigStoragePath(), 'files', 'Nam', 'QmdirectoryName', 'QmYfSQCgCwhxwYcdEwCkFJHicDe6rzCAb7AtLz3GrHmuU6')
+    assert.deepStrictEqual(fullPath, validPath)
+  })
+
+  it('Should fail if dirName and fileName are not passed into computeBasePath', function () {
+    try {
+      DiskManager.computeFilePathInDir()
+    } catch (e) {
+      assert.ok(e.message.includes('Must pass in valid dirName and fileName'))
+    }
+  })
+
+  /**
+   * extractCIDsFromFSPath
+   */
+  it('Should pass if extractCIDsFromFSPath is passed in a directory and file', function () {
+    const path = '/file_storage/files/QmYfSQCgCwhxwYcdEwCkFJHicDe6rzCAb7AtLz3Grouter/QmYfSQCgCwhxwYcdEwCkFJHicDe6rzCAb7AtLz3Grinner'
+    const matchObj = DiskManager.extractCIDsFromFSPath(path)
+    assert.deepStrictEqual(matchObj.isDir, true)
+    assert.deepStrictEqual(matchObj.outer, 'QmYfSQCgCwhxwYcdEwCkFJHicDe6rzCAb7AtLz3Grouter')
+    assert.deepStrictEqual(matchObj.inner, 'QmYfSQCgCwhxwYcdEwCkFJHicDe6rzCAb7AtLz3Grinner')
+  })
+
+  it('Should pass if extractCIDsFromFSPath is passed in just a file', function () {
+    const path = '/file_storage/files/QmYfSQCgCwhxwYcdEwCkFJHicDe6rzCAb7AtLz3Grinner'
+    const matchObj = DiskManager.extractCIDsFromFSPath(path)
+    assert.deepStrictEqual(matchObj.isDir, false)
+    assert.deepStrictEqual(matchObj.outer, 'QmYfSQCgCwhxwYcdEwCkFJHicDe6rzCAb7AtLz3Grinner')
+    assert.deepStrictEqual(matchObj.inner, null)
+  })
+
+  it('Should return null if extractCIDsFromFSPath is passed in no valid CID', function () {
+    const path = '/file_storage/files/QMcidhere'
+    const matchObj = DiskManager.extractCIDsFromFSPath(path)
+    assert.deepStrictEqual(matchObj, null)
```

Comment:
why did you choose to have it return null instead of throw error in this case? am concerned we might see some silent failures if we don't explicitly throw

Reply:
that's just the api i thought of. if you feel strongly i can change but didn't seem necessary to have a try/catch around this when we can just have an if check

Reply:
so we don't necessarily have to try-catch this in each caller since they're all wrapped in handleResponse anyway, my concern would be if the function occasionally returns null and we have no awareness. i guess why return null if there's no valid scenario where that should happen. throwing seems safer 🤷

Reply:
lol def ignore for now